### PR TITLE
MINOR: Tighten up metadata upgrade test

### PR DESCRIPTION
**Pull request by guozhangwang**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by guozhangwang** [comment]
Triggered https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2479/

File: streams/src/test/java/org/apache/kafka/streams/tests/SmokeTestUtil.java [hunk 1]
```diff
@@ -49,7 +49,7 @@
                     @Override
                     public void init(final ProcessorContext context) {
                         super.init(context);
-                        System.out.println("initializing processor: topic=" + topic + " taskId=" + context.taskId());
+                        System.out.println("[DEV] initializing processor: topic=" + topic + " taskId=" + context.taskId());
```
Comment (guozhangwang) [inline streams/src/test/java/org/apache/kafka/streams/tests/SmokeTestUtil.java:52]: These are just for easy debugging.

File: tests/kafkatest/tests/streams/streams_upgrade_test.py [hunk 2]
```diff
@@ -382,9 +382,9 @@ def start_all_nodes_with(self, version):
                 log_monitor.wait_until(kafka_version_str,
                                        timeout_sec=60,
                                        err_msg="Could not detect Kafka Streams version " + version + " " + str(node1.account))
-                monitor.wait_until("processed [0-9]* records from topic",
+                monitor.wait_until(self.processed_msg,
```
Comment (guozhangwang) [inline tests/kafkatest/tests/streams/streams_upgrade_test.py:385]: These are to cover cases where not exactly 100 records are processed.

**Comment by guozhangwang** [comment]
2479 succeeded, triggered https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2483/ again with repeating = 5.

**Comment by guozhangwang** [comment]
Call for review @bbejeck @vvcephei @ableegoldman

**Review by bbejeck — APPROVED** [review]
LGTM!

**Pull request merged**